### PR TITLE
Implement data generation script that will write data to a csv file

### PR DESCRIPTION
**Pull request by MeganKeesee-zz**
Write data generation script that: 
- creates 10M stock tickers
- generates a single row of data at a time for a csv
- opens a write stream that writes data to csv and drains data in the stream when full
- creates a csv to store 10M records

### CODE REVIEW

File: data-gen-sdc.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      }
+    } while (i > 0 && ok);
+    if (i > 0) {
+      // had to stop early, bucket full, write some more once it drains
+      writer.once('drain', write);
+    }
+  };
+};
+
+let writer = fs.createWriteStream('./dataTest1000000.csv');
```
Comment (RamKhalsa) [inline data-gen-sdc.js]: If you plan to change the number of records that it is writing, I'd recommend tying line 140, 163, and any other references to the same variable to allow you to specify that number in one place and have it update the function and filename!

File: data-gen-sdc.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+writer.write('Ticker, RecBuy, RecHold, RecSell, BuySum, SellSum \n');
+writeOneMillionTimes(writer, generateSingleRecord, null, () => console.log('Finished!'));
```
Comment (RamKhalsa) [inline data-gen-sdc.js]: Does a null argument here imply utf8?

Comment (RamKhalsa) [reply data-gen-sdc.js]: @MeganKeesee I would also be curious if you've found that you need to call "end" on the writeStream, or if it is automatically figuring out that you are done writing to it! I had some issues with this when doing gZip compression, where I had to call it manually.

File: package.json [hunk 3]
```diff
@@ -58,6 +58,8 @@
     "@babel/polyfill": "^7.4.0",
     "axios": "^0.18.0",
     "body-parser": "^1.18.3",
+    "csv-write-stream": "^2.0.0",
```
Comment (RamKhalsa) [inline package.json]: I see these new libraries added, but didn't see them reference in your files--are you using them?

**Review by RamKhalsa — APPROVED** [review]
@MeganKeesee looks great! I left a few comments; hopefully they're helpful!

File: data-gen-sdc.js (hunk 2 above)
Comment (MeganKeesee-zz) [reply data-gen-sdc.js]: > Does a null argument here imply utf8?

yes!

Comment (MeganKeesee-zz) [reply data-gen-sdc.js]: > @MeganKeesee I would also be curious if you've found that you need to call "end" on the writeStream, or if it is automatically figuring out that you are done writing to it! I had some issues with this when doing gZip compression, where I had to call it manually.

I haven't had any issues with it, but I'd love to chat to you about it in person to understand why you were having that issue.

File: package.json (hunk 3 above)
Comment (MeganKeesee-zz) [reply package.json]: Good eye. I was originally using these in my seed script, but decided not to in the end. I've deleted them from my package now!

**Pull request merged**